Features/21.2

AndreeaDan97 opened the pull request:
- creating two types of diagrams based on current created events and locations: the first one (events) shows a piechart containing  the percentage of available, validated and bought tickets of every created event, the second one (locations) depicts a barchart containing every location available with all its assigned events, each one of them having the available, validated and bought tickets percentage
- there are two separate pages for each type of diagrams

brehua committed:
Update translation.json

Added 2 more text labels for the new navigation tree

brehua committed:
Updated Main routes

Added a new Route to the new tab Statistics

brehua committed:
Created Stepper

Created a new stepper for the statistics page

brehua committed:
Create StatisticsPageSmart.tsx

Created the main page for the statistics

brehua committed:
Create EventStatisticsOverview.tsx

Overview component for the event statistic

brehua committed:
Updated Interface

brehua committed:
Create LocationStatisticsOverview.tsx

brehua committed:
Update package-lock.json

brehua committed:
Update package.json

AndreeaDan97 committed:
Generated a barchar diagram with mock data

brehua committed:
Upate Statistics page

AndreeaDan97 committed:
Merge branch 'features/21.2' of https://github.com/msg-CareerStart/event-management into features/21.2

AndreeaDan97 committed:
Events and Locations statistics pages put togheter

AndreeaDan97 committed:
Changed the props sent to Stepper

AndreeaDan97 committed:
Merge branch 'develop' of https://github.com/msg-CareerStart/event-management into features/21.2

brehua committed:
Created interfaces

Created interfaces for statistics for both location and events

brehua committed:
Update EventController.java

Missing import

AndreeaDan97 committed:
Locations' diagram: implementing the redux part

AndreeaDan97 committed:
Merge branch 'features/21.2' of https://github.com/msg-CareerStart/event-management into features/21.2

brehua committed:
Event StatisticPAge

brehua committed:
Update rootReducer.tsx

AndreiBrehuescu committed:
Update rootReducer.tsx

AndreeaDan97 committed:
some changes

AndreeaDan97 committed:
Merge branch 'features/21.2' of https://github.com/msg-CareerStart/event-management into features/21.2

brehua committed:
Update EventStatisticsOverview.tsx

Changed into displaying percentage

brehua committed:
Updated translation.json

Added language translation  for event and location statistics tab

brehua committed:
Update EventStatisticsOverview.tsx

Added a custom renderer

AndreeaDan97 committed:
Merge branch 'features/21.2' of https://github.com/msg-CareerStart/event-management into features/21.2

AndreeaDan97 committed:
Changes to redux part

- Everything works fine, besides one bug in the new Reducer

AndreeaDan97 committed:
Merge branch 'features/21.2' of https://github.com/msg-CareerStart/event-management into features/21.2

AndreeaDan97 committed:
fixed Reducer bug

AndreeaDan97 committed:
Locations' Statistics Diagram: finished

- little changes to logic, so that the code is more readable
- both types of diagrams work as they should

AndreeaDan97 committed:
Some changes to diagrams

- changed colors of locations' diagrams so it matches the requirements
- changed the percentage appearance (not it has "%" integrated)

AndreiSold commented on event-management-backend/src/main/java/ro/msg/event_management/controller/EventController.java:
```diff
@@ -297,7 +300,7 @@ public ResponseEntity options(HttpServletResponse response) {
 
     @GetMapping("/user/future")
-    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
+//    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
```
You don't need this here? If not completely remove it

AndreiSold commented on event-management-frontend/src/components/statisticsPage/StatisticsPageSmart.tsx:
```diff
@@ -0,0 +1,79 @@
+import { connect } from 'react-redux';
+import { fetchAllEvents, fetchAllExistingEvents } from '../../actions/EventsPageActions';
+import { AppState } from '../../store/store';
+
```
Remove this empty line, no need for it

AndreeaDan97 committed:
little adjustments to diagrams

AndreeaDan97 committed:
deleted unused function

brehua committed:
Added translation

Added translation for tooltip and drop down menu message when no items are selected.

brehua committed:
Update StepperStatistics.tsx

Added tooltip on the icons

brehua committed:
Update SearchBar.tsx

Fixed a previous bug on the searchbar
